samples/python/evaluate_qrcode/evaluate_qrcode.py

```diff
@@ -0,0 +1,386 @@
+#!/usr/bin/env python
+
+'''
+
+This sample uses dataset from
+http://www.fit.vutbr.cz/research/groups/graph/pclines/pub_page.php?id=2012-SCCG-QRtiles
+and markup files located in markup/ folder
+for evaluation QR-code detection algorithm
+
+'''
+
+# Python 2/3 compatibility
+import optparse
+import os
+import sys
+from os.path import join
+
+import cv2
+import numpy as np
+import json
+
+
+def main():
+    parser = optparse.OptionParser()
+    parser.add_option('--images', default="qrcode-datasets/datasets", help="Location of directory with input images")
+    parser.add_option('--data', default="markup", help="Location of directory with input data")
+    options, arguments = parser.parse_args()
+
+    path_to_images = options.images
+    path_to_data = options.data
+
+    dir_images = [f for f in sorted(os.listdir(path_to_images))]
+    dir_data = [f for f in sorted(os.listdir(path_to_data))]
+
+    run_metric_through_categories(path_to_images, dir_images, path_to_data, dir_data, 1)
+    run_metric_through_categories(path_to_images, dir_images, path_to_data, dir_data, 2)
+    run_metric_through_categories(path_to_images, dir_images, path_to_data, dir_data, 3)
+    run_metric_through_categories(path_to_images, dir_images, path_to_data, dir_data, 4)
+    run_metric_through_categories(path_to_images, dir_images, path_to_data, dir_data, 5)
+
+
+def run_metric_through_categories(path_to_images, categories, path_to_data, dir_data, metric):
+    print_metric_description(metric)
+
+    detection_stat = []
+    decoding_stat = []
+    scores = []
+
+    qrDetector = cv2.QRCodeDetector()
+
+    for idx, category in enumerate(categories):
+        category_score = []
+
+        detected_in_category = 0
+        decoded_in_category = 0
+
+        path_to_category = os.path.join(path_to_images, category)
+        number_qrcodes_in_category = get_number_qrcodes_in_category(path_to_data, dir_data[idx])
+
+        input_images = get_input_images(path_to_category)
+        input_data = get_input_data(path_to_data, dir_data[idx])
+
+        total_number_images = len(input_data['test_images'])
```

Comment:
what kind of type does `input_data` have? if `np.array` then please use `input_data.size`